client/libra_wallet/src/mnemonic.rs

```diff
@@ -33,9 +48,10 @@ impl Mnemonic {
     /// Generate mnemonic from string
     pub fn from(s: &str) -> Result<Mnemonic> {
```

Comment:
It's quite troubling to me that we're not using the verification of validity of the checksum here to early-fail in case the user has provided a mnemonic with a typo.

This would involve comparing the checksum at the end of the mnemonic to one recomputed from the present function's input in a manner similar to this:
https://github.com/libra/libra/blob/aa723b1e4dbb2c95e514bfb13c108527b9cdd5bc/client/libra_wallet/src/mnemonic.rs#L63-L65

Reply:
Good point, it's true I didn't change the original implementation specifics, mainly focused on entropy incompatibility issues, but I will take the opportunity and fix this as well.